lib/formalist/child_forms/builder.rb

```diff
@@ -0,0 +1,45 @@
+require "json"
+require_relative "child_form"
+
+module Formalist
+  module ChildForms
+    class Builder
+      attr_reader :embedded_forms
+
+      def initialize(embedded_form_collection)
+        @embedded_forms = embedded_form_collection
+      end
+
+      def call(input)
+        return input if input.nil?
+
+        input = input.is_a?(String) ? JSON.parse(input) : input
+
+        input.map { |node| visit(node) }
+      end
+      alias_method :[], :call
+
+      private
+
+      def visit(node)
+        name, data = node.values_at(:name, :data)
+
+        embedded_form = embedded_forms[name]
```

Comment:
What would happen here if `embedded_form` turned out to be `nil`, i.e. the right kind of form isn't available within the collection? Would we want to raise a specific error here rather the generic "undefined method on nilclass" error being raised from within `child_form`?

Reply:
Thanks for this pickup. You are right, we need a way to rescue from an error here if a form is removed from a list of available forms but the data still exists in the database. Good chance that could happen if existing data isn't migrated.